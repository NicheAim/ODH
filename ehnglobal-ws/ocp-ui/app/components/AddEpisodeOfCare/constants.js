/*
 *
 * Episode of Care  constants
 *
 */
export const ENTERED_IN_ERROR_STATUS = 'entered-in-error';
