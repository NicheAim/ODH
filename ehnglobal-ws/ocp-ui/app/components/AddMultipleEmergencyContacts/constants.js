export const TEXT_MIN_LENGTH = 3;
