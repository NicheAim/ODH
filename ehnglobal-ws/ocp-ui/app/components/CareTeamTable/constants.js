
/**
 *
 * @type {string}
 */
export const EXPANDED_TABLE_COLUMNS = '.5fr 1fr 1fr 1fr 1fr 2fr 1fr 1fr 1fr .5fr';
export const SUMMARIZED_TABLE_COLUMNS = 'repeat(5, 0.75fr) 0.75fr';
