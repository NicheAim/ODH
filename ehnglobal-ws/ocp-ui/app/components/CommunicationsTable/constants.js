
export const EXPANDED_TABLE_COLUMNS = '.5fr 1.5fr 1fr 1.5fr repeat(6, 1fr) 1fr';
export const SUMMARIZED_TABLE_COLUMNS = '.5fr 3fr 2fr 2fr 1fr';

export const SUMMARY_VIEW_WIDTH = 1000; // The width of the panel in pixels
