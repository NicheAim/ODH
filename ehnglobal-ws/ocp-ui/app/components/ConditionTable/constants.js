// export const CONDITION_TABLE_COLUMNS = '1fr 1fr 1fr 50px';

export const CONDITION_TABLE_COLUMNS = '1.5fr 1fr 1fr .6fr';
