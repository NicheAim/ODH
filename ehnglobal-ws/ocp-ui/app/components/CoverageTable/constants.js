
export const EXPANDED_TABLE_COLUMNS = 'repeat(9, 1fr) 50px';
export const SUMMARIZED_TABLE_COLUMNS = '50px 3fr 1fr 2fr';

export const SUMMARY_VIEW_WIDTH = 695; // The width of the panel in pixels
