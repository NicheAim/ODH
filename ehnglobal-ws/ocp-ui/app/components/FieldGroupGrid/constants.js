export const PREFIX = 'fieldGroupPrefix';
export const MAIN = 'fieldGroupMain';
export const SUFFIX = 'fieldGroupSuffix';
