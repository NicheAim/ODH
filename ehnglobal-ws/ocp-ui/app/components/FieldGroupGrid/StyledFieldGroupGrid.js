import PropTypes from 'prop-types';
import { Grid } from 'styled-css-grid';
import { PREFIX, MAIN, SUFFIX } from './constants';

const StyledFieldGroupGrid = Grid.extend`
  grid-template-columns: 1fr;
  grid-template-areas: "${PREFIX}" "${MAIN}"${(props) => props.withSuffix ? ` "${SUFFIX}"` : ''};

  @media (min-width: 768px) {
    grid-template-columns: ${(props) => props.withSuffix ? '1fr 2fr 1fr' : '1fr 2fr'};
    grid-template-areas: "${PREFIX} ${MAIN}${({ withSuffix }) => withSuffix ? ` ${SUFFIX}` : ''}";
  }
`;

StyledFieldGroupGrid.propTypes = {
  ...Grid.propTypes,
  withSuffix: PropTypes.bool,
};

StyledFieldGroupGrid.defaultProps = {
  withSuffix: false,
};

export default StyledFieldGroupGrid;
