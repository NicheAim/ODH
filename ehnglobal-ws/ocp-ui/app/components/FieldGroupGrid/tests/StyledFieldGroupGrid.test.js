import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import 'jest-styled-components';

import StyledFieldGroupGrid from '../StyledFieldGroupGrid';
import { PREFIX, MAIN, SUFFIX } from '../constants';

configure({ adapter: new Adapter() });

describe('<StyledFieldGroupGrid />', () => {
  describe('snapshot tests', () => {
    it('should match snapshot', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<StyledFieldGroupGrid>{children}</StyledFieldGroupGrid>);

      // Assert
      expect(renderedComponent).toMatchSnapshot();
    });

    it('should match snapshot withSuffix', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<StyledFieldGroupGrid withSuffix>{children}</StyledFieldGroupGrid>);

      // Assert
      expect(renderedComponent).toMatchSnapshot();
    });
  });

  describe('structural tests', () => {
    it('should have children', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<StyledFieldGroupGrid>{children}</StyledFieldGroupGrid>);

      // Assert
      expect(renderedComponent.contains(children)).toEqual(true);
    });

    it('should have children withSuffix', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<StyledFieldGroupGrid withSuffix>{children}</StyledFieldGroupGrid>);

      // Assert
      expect(renderedComponent.contains(children)).toEqual(true);
    });
  });

  describe('style tests', () => {
    it('should be grid', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<StyledFieldGroupGrid>{children}</StyledFieldGroupGrid>);

      // Assert
      expect(renderedComponent).toHaveStyleRule('display', 'grid');
    });

    it('should be grid withSuffix', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<StyledFieldGroupGrid withSuffix>{children}</StyledFieldGroupGrid>);

      // Assert
      expect(renderedComponent).toHaveStyleRule('display', 'grid');
    });

    it('should have default styles', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<StyledFieldGroupGrid>{children}</StyledFieldGroupGrid>);

      // Assert
      expect(renderedComponent).toHaveStyleRule('grid-template-columns', '1fr');
      expect(renderedComponent).toHaveStyleRule('grid-template-areas', `"${PREFIX}" "${MAIN}"`);
    });

    it('should have default styles withSuffix', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<StyledFieldGroupGrid withSuffix>{children}</StyledFieldGroupGrid>);

      // Assert
      expect(renderedComponent).toHaveStyleRule('grid-template-columns', '1fr');
      expect(renderedComponent).toHaveStyleRule('grid-template-areas', `"${PREFIX}" "${MAIN}" "${SUFFIX}"`);
    });

    it('should have styles in min-width: 768px', () => {
      // Arrange
      const children = <div>test</div>;
      const media = '(min-width: 768px)';

      // Act
      const renderedComponent = shallow(<StyledFieldGroupGrid>{children}</StyledFieldGroupGrid>);

      // Assert
      expect(renderedComponent).toHaveStyleRule('grid-template-columns', '1fr 2fr', { media });
      expect(renderedComponent).toHaveStyleRule('grid-template-areas', `"${PREFIX} ${MAIN}"`, { media });
    });

    it('should have styles withSuffix in min-width: 768px', () => {
      // Arrange
      const children = <div>test</div>;
      const media = '(min-width: 768px)';

      // Act
      const renderedComponent = shallow(<StyledFieldGroupGrid withSuffix>{children}</StyledFieldGroupGrid>);

      // Assert
      expect(renderedComponent).toHaveStyleRule('grid-template-columns', '1fr 2fr 1fr', { media });
      expect(renderedComponent).toHaveStyleRule('grid-template-areas', `"${PREFIX} ${MAIN} ${SUFFIX}"`, { media });
    });
  });
});
