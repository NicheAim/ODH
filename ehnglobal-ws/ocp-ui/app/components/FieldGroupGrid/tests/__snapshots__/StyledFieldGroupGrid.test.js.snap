// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledFieldGroupGrid /> snapshot tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div
    withSuffix={false}
  >
    <div>
      test
    </div>
  </styled.div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        test
      </div>,
      "className": "sc-bdVaJa jGCfRb",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test",
      },
      "ref": null,
      "rendered": "test",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          test
        </div>,
        "className": "sc-bdVaJa jGCfRb",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<StyledFieldGroupGrid /> snapshot tests should match snapshot withSuffix 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div
    withSuffix={true}
  >
    <div>
      test
    </div>
  </styled.div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        test
      </div>,
      "className": "sc-bdVaJa jiqbbt",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test",
      },
      "ref": null,
      "rendered": "test",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          test
        </div>,
        "className": "sc-bdVaJa jiqbbt",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
